### PR TITLE
Alert Component

### PR DESCRIPTION
**Pull request by nicko-winner**

### CODE REVIEW

File: src/Alert/index.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import React, { useRef, useEffect} from 'react';
+import PropTypes from 'prop-types';
+
+const Alert = (props) => {
```
[the remark this answers is not in the document]
Comment (nicko-winner) [reply src/Alert/index.js:4]: It's my really a subjective take on beauty..it always looked strange to me to type the same variable name in the function signature, the props types, and when you use it, and I never really liked function signatures that span multiple lines. But typing `props` 10 times some how better?  for me but, but not by a long shot so i'm not married to this style at all.

File: src/Alert/index.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            <hx-alert
+                type={props.type}
+                status={props.status}
+                persist={props.persist}
+                cta={props.cta}
+                className={props.className}
+                id={props.id}
+                ref={hxRef}
+            >
+                {' '}{props.children}
```
[the remark this answers is not in the document]
Comment (nicko-winner) [reply src/Alert/index.js:31]: JSX removes all the spaces between elements, event when they span across multiple lines. Some times helix components rely on the natural spacing added by having a an element spread over multiple lines. So we are manually adding a space to compensate for JSX removing the the space. This space was added to compensate for an older version of helix I will check if its still necessary.

File: src/Alert/index.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        hxRef.current.addEventListener('open', props.onOpen);
+        hxRef.current.addEventListener('close', props.onClose);
+        return () => {
+            hxRef.current.removeEventListener('open', props.onOpen);
+            hxRef.current.removeEventListener('close', props.onClose);
+        };
+    }, []);
+
+    return (
+        <>
```
[the remark this answers is not in the document]
Comment (nicko-winner) [reply src/Alert/index.js:17]: hmm, I wonder if github fixed it? our team has been late adopters of React 16, and have not seen this issue yet. Keep going with it until we see an issue?

File: src/Alert/index.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+Alert.propTypes = {
+    className: PropTypes.string,
+    children: PropTypes.node,
+    type: PropTypes.string,
+    status: PropTypes.string,
+    cta: PropTypes.string,
+    onOpen: PropTypes.func,
+    onClose: PropTypes.func
```
[the remark this answers is not in the document]
Comment (nicko-winner) [reply src/Alert/index.js:44]: I think just children may be the only required item.

File: src/Alert/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (michaelmang) [reply src/Alert/index.js:4]: That's cool, I like it to but haven't seen it before. Good to know!

File: src/Alert/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (michaelmang) [reply src/Alert/index.js:17]: I think it's fine to leave it as you have it. I was curious 👍

File: src/Alert/index.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (michaelmang) [reply src/Alert/index.js:44]: I think you are correct. Also, I noticed that the `persist` prop is missing from the `propTypes`.

Do we want to default the `persist` and `type` props?

[the remark this answers is not in the document]
Comment (nicko-winner) [reply src/Alert/index.js:44]: good catch. i'm tempted to not use default propTypes and just let helix-ui pick the defaults. That way our wrapper defaults wont get in the way of what Helix may set as defaults in the future. (makes the wrapper slightly more light weight). @100stacks you have an any opinion on if we should be setting defaults props/attributes before passing them on to helix?

File: src/Alert/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicko-winner) [reply src/Alert/index.js]: removed the extra space. Alert still looks good without it.

**Comment by 100stacks** [comment]
Fixes #19 (`<Alert/>`)

File: src/Alert/index.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (100stacks) [reply src/Alert/index.js:45]: @HelixDesignSystem/helix-react-core I think we left this open during our sync today.  Balancing explicit defaults vs. implied HelixUI defaults. 

We can address again as needed.

**Pull request merged**